### PR TITLE
Try: Translation Scanner

### PR DESCRIPTION
**Pull request by deBhal**
This PR adds a "Translation Scanner" to calypso to forward translations to the new `https://translate.wordpress.com/projects/wpcom/${locale}/default/recording/` tool.

Requires code-D28783 to provide API endpoints, see that diff for testing instructions.

### CODE REVIEW

**Comment by deBhal** [comment]
I've added some logic to watch the `gp-record` for cookies, but it's not integrated with the UI, as I'm not sure where it's going to live yet.

I'm leaning towards putting it into the community-translator at the moment, so I've gone with the low-frills object mode and skipped the UI for now, so you'll need to fiddle with the console to see it working, but you can:
1) `document.cookie='gp-record=1;path=/;domain=calypso.localhost';`
2) Navigate around to trigger some translations
3) `document.cookie='gp-record=;path=/;domain=calypso.localhost;expires=Thu, 01 Jan 1970 00:00:01 GMT';`
4) check `window.i18nScanner.loggedTranslations`

**Comment by akirk** [comment]
After some back and forth whether this could be done as a side-effect of the community translator, we decided to move forward on this in with a light-weight version (because eventually we'll want to retire the jQuery community translator in favor of an always-on #21591).

o Remove UI (starting or stopping happens on translate.wordpress.com)
o Verify it works logged out (since it will be activated through the cookie)
o Ping `query-by-originals` with the strings instead of making the Community Translator do it

File: client/layout/community-translator/launcher.jsx
Comment (deBhal) [inline client/layout/community-translator/launcher.jsx]: I'm not sure entirely sure about this. It's nice to be able to catch all the startup translations, and this does get a lot of them, but I'm not sure if it's early enough to catch all of them.

File: client/lib/i18n-utils/translation-scanner.js
Comment (lsl) [inline client/lib/i18n-utils/translation-scanner.js]: move check inside `install()`

File: client/lib/i18n-utils/translation-scanner.js
Comment (lsl) [inline client/lib/i18n-utils/translation-scanner.js]: remove - handled in community translator

File: client/lib/i18n-utils/translation-scanner.js
Comment (lsl) [inline client/lib/i18n-utils/translation-scanner.js]: remove - handled in community translator

**Review by lsl — CHANGES_REQUESTED** [review]
Looks good so far, there are a few things you should probably remove before merging.

Still couldn't get it working on my end, I'll have another go tomorrow a.m. and see if I can get it to kick with fresh eye's.

**Comment by lsl** [comment]
Got this working now, works great. - needed manually set cookie for localhost testing.

**Review by lsl — APPROVED** [review]
Approving to unblock, tested and LGTM. Please clean up the extra checks before merging.

File: client/lib/i18n-utils/constants.js
Comment (deBhal) [reply client/lib/i18n-utils/constants.js]: No. Good catch, looks like this is broken. I'm going to plead "grandfathered" https://github.com/Automattic/wp-calypso/issues/33623 but I think it actually impacts the v2 community translator

File: client/lib/i18n-utils/translation-scanner.js
Comment (deBhal) [reply client/lib/i18n-utils/translation-scanner.js]: I'll remove the `install()` in `start()` instead. `install()` puts the translation filter and the cookie watcher into place, and then the cookie watcher is the only thing calling start/stop at the moment so it seems reasonable (for now at least) to assume installation in `start()`

**Pull request merged**
